src/routes/dockerflow.js

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+// In this router we implement the services required by the dockerflow mozilla
+// best practices, as described in https://github.com/mozilla-services/Dockerflow.
+
+import Router from '@koa/router';
+import fs from 'fs';
+import path from 'path';
+
+export default function() {
+  const router = new Router();
+
+  // "Respond to /__version__ with the contents of /app/version.json."
+  // This file is generated by the script bin/generate-verison-file.js at build
+  // time in CircleCI and lives in dist/version.json.
+  // We serve it directly if present, otherwise returns a 404.
+  router.get('/__version__', async ctx => {
+    // We use path.join with __dirname so that the path doesn't depend on the
+    // current working directory.
+    // And we prefer it over require.resolve or path.resolve because they're not
+    // easy to mock in tests (maybe impossible). Also path.join does no
+    // filesystem access so it's probably faster than them.
+    const fileName = path.join(__dirname, '../../dist/version.json');
+
+    try {
+      const content = await fs.promises.readFile(fileName);
+      ctx.body = content;
+      ctx.type = 'application/json';
+    } catch (e) {
+      if (e.code === 'ENOENT') {
```

Comment:
It'd probably be good to document this code. I'm not sure what it is without looking it up.

`ENOENT`.

Also, I think if this is not found it should be a 500, right? 404 is if the end user mistyped something. Here we're internally not configured.